sunpy/map/tests/test_mapsequence.py

```diff
@@ -147,6 +165,25 @@ def test_all_meta(mapsequence_all_the_same):
     assert np.all(np.asarray([meta[i] == mapsequence_all_the_same[i].meta for i in range(0, len(meta))]))
 
 
+def test_repr(mapsequence_all_the_same, mapsequence_different_maps):
+    """
+    Tests that overidden __repr__ functionality works as expected. Test
+    for mapsequence of same maps as well that of different maps.
+    """
+    # Test the case of MapSequence having same maps
+    expected_out = f'MapSequence of 2 elements, with maps from AIAMap'
+    obtained_out = repr(mapsequence_all_the_same)
+    assert len(mapsequence_all_the_same) == 2
+    assert obtained_out == expected_out
+
+    # Test the case of MapSequence having different maps
+    expected_out1 = f'MapSequence of 2 elements, with maps from AIAMap, EITMap'
+    expected_out2 = f'MapSequence of 2 elements, with maps from EITMap, AIAMap'
+    obtained_out = repr(mapsequence_different_maps)
+    assert len(mapsequence_different_maps) == 2
+    assert obtained_out == expected_out1 or obtained_out == expected_out2
```

Comment:
Why would the repr out be `expected_out1` or `expected_out2`? Should it be fixed?

Reply:
We are adding the class names of the maps present in a set, which is unordered. Therefore, in this case, the output can be either of them.